File: src/payloads/workspace.ts
export interface WorkspaceResponse {
  id: string;
  name: string;
  slug: string;
  logo_url?: string | null;
  owner_id: string;
  created_at: string;
}

export interface InviteDetailsResponse {
  workspace_name: string;
  workspace_slug: string;
  role: string;
  email_restricted: boolean;
}

export interface JoinInviteResponse {
  workspace_id: string;
  workspace_slug: string;
  already_member: boolean;
}

File: rust/athene-contracts/src/workspace.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, ToSchema)]
pub struct WorkspaceResponse {
    pub id: String,
    pub name: String,
    pub slug: String,
    pub logo_url: Option<String>,
    pub owner_id: String,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, ToSchema)]
pub struct InviteDetailsResponse {
    pub workspace_name: String,
    pub workspace_slug: String,
    pub role: String,
    pub email_restricted: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, ToSchema)]
pub struct JoinInviteResponse {
    pub workspace_id: String,
    pub workspace_slug: String,
    pub already_member: bool,
}

pub type WorkspaceListResponse = crate::ListResponse<WorkspaceResponse>;
